airflow-core/src/airflow/cli/commands/config_command.py

```diff
@@ -704,3 +705,148 @@ def lint_config(args) -> None:
         console.print("\n[red]Please update your configuration file accordingly.[/red]")
     else:
         console.print("[green]No issues found in your airflow.cfg. It is ready for Airflow 3![/green]")
+
+
+@providers_configuration_loaded
+def update_config(args) -> None:
+    """
+    Update the airflow.cfg file to migrate configuration changes from Airflow 2.x to Airflow 3.
+
+    This command scans the current configuration file for parameters that have been renamed,
+    removed, or had their default values changed in Airflow 3.0, and automatically updates
+    both the configuration file and relevant environment variables accordingly.
+
+    CLI Arguments:
+        --section: str (optional)
+            Comma-separated list of configuration sections to update.
+            Example: --section core,database
+
+        --option: str (optional)
+            Comma-separated list of configuration options to update.
+            Example: --option sql_alchemy_conn,dag_concurrency
+
+        --ignore-section: str (optional)
+            Comma-separated list of configuration sections to ignore during update.
+            Example: --ignore-section webserver
+
+        --ignore-option: str (optional)
+            Comma-separated list of configuration options to ignore during update.
+            Example: --ignore-option check_slas
+
+        --verbose: flag (optional)
+            Enable detailed output, including a summary of changes and any ignored items.
+            Example: --verbose
+
+    Examples:
+        1. Update the entire configuration file:
+            airflow config update
+
+        2. Update only the 'core' and 'database' sections:
+            airflow config update --section core,database
+
+        3. Update only specific options:
+            airflow config update --option sql_alchemy_conn,dag_concurrency
+
+        4. Ignore updates for a specific section:
+            airflow config update --ignore-section webserver
+
+        5. Enable verbose output for more detailed information:
+            airflow config update --verbose
+
+    :param args: The CLI arguments for updating configuration.
+    """
+    console = AirflowConsole()
+    changes_applied = []
+
+    update_sections = args.section if args.section else None
+    update_options = args.option if args.option else None
+    ignore_sections = args.ignore_section if args.ignore_section else []
+    ignore_options = args.ignore_option if args.ignore_option else []
+
+    for change in CONFIGS_CHANGES:
+        conf_section = change.config.section
+        conf_option = change.config.option
+
+        if update_sections is not None and conf_section not in update_sections:
+            continue
+        if update_options is not None and conf_option not in update_options:
```

Comment:
options should be scoped to a section too

Reply:
Do you mean something like following:
```
if update_options is not None:
            if update_sections is not None:
                full_option = f"{conf_section}.{conf_option}"
                if full_option not in update_options:
                    continue
            else:
                if conf_option not in update_options:
                    continue
```

Reply:
I mean option names aren't unique, section name + option name is.

E.g. there is `[metrics] otel_on` and `[traces] otel_on`. Accepting just `otel_on` makes it awkward.

Reply:
Changed it.